### ISSUE TITLE
VavrOptionMapper always uses ColumnMapper

### ISSUE THREAD

**Issue by martyn0ff** [jdbi/jdbi#2558]
I've got an issue where `VavrOptionMapper` is not recognizing my existing `ConstructorMapper` for `Foo`.

```java
org.jdbi.v3.core.mapper.NoSuchMapperException: No mapper for type class com.acme.Foo nested in Option
	at org.jdbi.v3.vavr.VavrOptionMapper.lambda$map$2(VavrOptionMapper.java:54)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at org.jdbi.v3.vavr.VavrOptionMapper.map(VavrOptionMapper.java:54)
	at org.jdbi.v3.vavr.VavrOptionMapper.map(VavrOptionMapper.java:30)
	at org.jdbi.v3.core.mapper.SingleColumnMapper.lambda$new$0(SingleColumnMapper.java:41)
	at org.jdbi.v3.core.mapper.SingleColumnMapper.map(SingleColumnMapper.java:55)
	at org.jdbi.v3.core.result.internal.ResultSetResultIterator.next(ResultSetResultIterator.java:83)
	at org.jdbi.v3.core.result.ResultIterable.findOne(ResultIterable.java:198)
```

Turns out that this mapper assumes that the mapping should always be done by the means of `ColumnMapper`:
https://github.com/jdbi/jdbi/blob/6456b974f9acbdc8cc04c84ac87be63b2ca809df/vavr/src/main/java/org/jdbi/v3/vavr/VavrOptionMapper.java#L53

I believe that it should instead use a call to `findMapperFor(GenericType)` instead. For the time being this is easily fixable by using:
```java
...
.mapTo(Foo.class)
.map(Option::of)
...
```

instead of

```java
...
.mapTo(new GenericType<Option<Foo>>() {})
...
```

**Comment by hgschmie** [comment]
Hi @martyn0ff,

Thank you for filing an issue with Jdbi. This is a shortcoming of Jdbi as we do not currently have a row factory for vavr `Option`. This will be supported in the next release.

**Comment by hgschmie** [comment]
fixed in 3.43.0
